### PR TITLE
 support of adding _timestamp in during mapping

### PR DESCRIPTION
**Pull request by savdeep**
Added proxy methods for _timestamp

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ All notable changes to this project will be documented in this file based on the
 - Support for field_value_factor #953
 - Added setMinDocCount and setExtendedBounds options #947
 - Avoid environment dependecies in tests #938
+- Added support for intialise mapping with _timestamp
```
Comment (ewgRa) [inline CHANGELOG.md]: can you add reference to #964?

File: lib/Elastica/Type.php [hunk 2]
```diff
@@ -299,6 +299,22 @@ public function setMapping($mapping)
     }
 
     /**
+     * Sets value type mapping for this type with timestamp .
+     *
+     * @param \Elastica\Type\Mapping|array $mapping Elastica\Type\MappingType object or property array with all mappings
+     *
+     * @return \Elastica\Response
+     */
+    public function setMappingWithTimeStamp($mapping,$timestamp)
```
Comment (ewgRa) [inline lib/Elastica/Type.php]: I'm not sure is this method needed? Mapping know about timestamp, Type know only about mapping, but not mapping details.

File: lib/Elastica/Type/Mapping.php [hunk 3]
```diff
@@ -294,4 +294,40 @@ public static function create($mapping)
...
+    }
+
+    /**
+     * Enables _timestamp for all documents in this type.
+     *
+     * @param bool $enabled OPTIONAL (default = true)
+     *
+     * @return $this
+     */
+    public function enableTimestamp($enabled = true)
```
Comment (ewgRa) [inline lib/Elastica/Type/Mapping.php]: not sure is this needed?

File: lib/Elastica/Type/Mapping.php [hunk 4]
```diff
@@ -294,4 +294,40 @@ public static function create($mapping)
...
+    }
+
+    /**
+     * Enables and stores _timestamp for all documents in this type.
+     *
+     * @param string $store OPTIONAL (default = "yes")
+     *
+     * @return $this
+     */
+    public function storeTimestamp($store = "yes")
```
Comment (ewgRa) [inline lib/Elastica/Type/Mapping.php]: not sure is this needed?

**Comment by ewgRa** [comment]
Would be nice if you add some tests

File: CHANGELOG.md (hunk 1 above)
Comment (savdeep) [inline CHANGELOG.md]: Of course

File: lib/Elastica/Type.php (hunk 2 above)
Comment (savdeep) [inline lib/Elastica/Type.php]: Yes, type should know about timestamp. But in type class there isn't any way of referencing mapping directly (on members). In setMapping method send is called immediately after making mapping, so i needed to set timestamp param before that. There is no room to update type w.r.t  timestamp.

File: lib/Elastica/Type/Mapping.php (hunk 3 above)
Comment (savdeep) [inline lib/Elastica/Type/Mapping.php]: setTimestamp is needed, but rest i added for other persons perusal

**Comment by savdeep** [comment]
Tests, absolutely but after verifying above changes.

File: lib/Elastica/Type.php (hunk 2 above)
Comment (ruflin) [inline lib/Elastica/Type.php]: Instead of introducing a new function, I would suggest to add a param to setMapping

Comment (ewgRa) [inline lib/Elastica/Type.php]: Disagree. _timestamp it is a mapping this. There is no reason to introduce it in Type. And what about another magic mapping fields like _size, _index... etc.? Also add as param?

**Comment by savdeep** [comment]
I will do some changes and get back to you

**Pull request closed without merging**